### PR TITLE
Implement Gen.char() - Closes #1005

### LINKED ISSUE #1005: Proposal: Gen.char()
It would be great if `Gen.char()` existed. Looks like it would be fairly trivial to implement, with `random()` of the `Gen<Char>` returning `Random.nextPrintableChar()`.

### PR DESCRIPTION
**Pull request by EarthCitizen**
Wrote one test to verify that seed value creates reproducible results. Writing a test that verifies different outcomes when no seed is present would be flakey with a good chance of failure.

### CODE REVIEW

File: kotlintest-assertions/src/commonMain/kotlin/io/kotlintest/properties/gens.kt [hunk 1]
```diff
@@ -242,6 +242,17 @@ fun Gen.Companion.bool(): Gen<Boolean> = object : Gen<Boolean> {
...
 }
 
+/**
+ * Returns a stream of randomly-chosen Chars
+ */
+fun Gen.Companion.char(): Gen<Char> = object : Gen<Char> {
+  override fun constants(): Iterable<Char> = emptyList()
+  override fun random(seed: Long?): Sequence<Char> {
+    val r = if (seed == null) Random.Default else Random(seed)
+    return generateSequence { r.nextPrintableChar() }
```
Comment (LeoColman) [inline kotlintest-assertions/src/commonMain/kotlin/io/kotlintest/properties/gens.kt:252]: I'm wondering if we want to limit ourselves to the "Printable Chars" only, or if we want to generate many other chars. What do you think?

File: kotlintest-assertions/src/jvmTest/kotlin/com/sksamuel/kotlintest/properties/CharGenTest.kt [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package com.sksamuel.kotlintest.properties
+
+import io.kotlintest.properties.Gen
+import io.kotlintest.properties.char
+import io.kotlintest.shouldBe
+import io.kotlintest.specs.FunSpec
+
+class CharGenTest : FunSpec({
```
Comment (LeoColman) [inline kotlintest-assertions/src/jvmTest/kotlin/com/sksamuel/kotlintest/properties/CharGenTest.kt:8]: Perhaps adding an exhaustive test that verifies all possible characters are obtained?

**Comment by sksamuel** [comment]
Have two gens. Gen char and gen printable char

**Comment by EarthCitizen** [comment]
I wanted to propose another idea, so I was trying to experiment with something, but I am having trouble figuring out how to get `kotlin.text.CharCategory` onto the classpath of `io.kotlintest.properties.gens`.

It is definitely in `kotlin-stdlib-1.3.50.jar`. I downloaded it and looked inside. Here is what I tried to get it onto the classpath:

In `kotlintest-assertions/build.gradle`:

- Added `implementation 'org.jetbrains.kotlin:kotlin-stdlib:1.3.50'` under dependencies in root, commonMain, and jvmMain
- Added `implementation kotlin("stdlib")` under dependencies in commonMain and jvmMain

In the Kotlin REPL, importing this class works fine:
```
>>> import kotlin.text.CharCategory
>>> CharCategory.UPPERCASE_LETTER.contains('T')
res38: kotlin.Boolean = true
```
Despite this, the build does not find `kotlin.text.CharCategory`. Any ideas?

File: kotlintest-assertions/src/commonMain/kotlin/io/kotlintest/properties/gens.kt (hunk 1 above)
Comment (EarthCitizen) [reply kotlintest-assertions/src/commonMain/kotlin/io/kotlintest/properties/gens.kt:252]: I think that is a good idea. I am working on another proposal.

**Comment by LeoColman** [comment]
Probably because you need to upgrade all the kotlin dependencies to use 1.3.50.
And that's ok!

**Comment by sksamuel** [comment]
Stdlib should already be on the classpath. And for jvmMain you need
stdlib-jdk8

**Comment by EarthCitizen** [comment]
@sksamuel I just noticed that CharCategory is not available for JS. I am new to Kotlin and I am slowly getting used to the complexity of multiple platforms. This was the cause of the error.

**Comment by EarthCitizen** [comment]
I have another proposal for this, so I am going to close and submit another PR.

**Pull request closed without merging**